aws-secretsmanager/src/main/java/io/micronaut/aws/secretsmanager/SecretsManagerGroupNameAwareKeyValueFetcher.java

```diff
@@ -0,0 +1,187 @@
+ * limitations under the License.
+ */
+package io.micronaut.aws.secretsmanager;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.micronaut.aws.distributedconfiguration.GroupNameAwareKeyValueFetcher;
+import io.micronaut.context.annotation.BootstrapContextCompatible;
+import io.micronaut.context.annotation.Requires;
+import io.micronaut.core.annotation.Experimental;
+import io.micronaut.core.annotation.NonNull;
+import jakarta.inject.Singleton;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import software.amazon.awssdk.services.secretsmanager.SecretsManagerClient;
+import software.amazon.awssdk.services.secretsmanager.model.DecryptionFailureException;
+import software.amazon.awssdk.services.secretsmanager.model.Filter;
+import software.amazon.awssdk.services.secretsmanager.model.FilterNameStringType;
+import software.amazon.awssdk.services.secretsmanager.model.GetSecretValueRequest;
+import software.amazon.awssdk.services.secretsmanager.model.GetSecretValueResponse;
+import software.amazon.awssdk.services.secretsmanager.model.InternalServiceErrorException;
+import software.amazon.awssdk.services.secretsmanager.model.InvalidParameterException;
+import software.amazon.awssdk.services.secretsmanager.model.InvalidRequestException;
+import software.amazon.awssdk.services.secretsmanager.model.ListSecretsRequest;
+import software.amazon.awssdk.services.secretsmanager.model.ListSecretsResponse;
+import software.amazon.awssdk.services.secretsmanager.model.ResourceNotFoundException;
+import software.amazon.awssdk.services.secretsmanager.model.SecretListEntry;
+import software.amazon.awssdk.services.secretsmanager.model.SecretsManagerException;
+
+import java.util.Base64;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+/**
+ * {@link GroupNameAwareKeyValueFetcher} implementations for AWS Secrets Manager.
+ *
+ * @author sbodvanski
+ * @since 3.8.0
+ */
+@Experimental
+@Requires(beans = {SecretsManagerClient.class})
+@BootstrapContextCompatible
+@Singleton
+public class SecretsManagerGroupNameAwareKeyValueFetcher implements GroupNameAwareKeyValueFetcher {
+    private static final Logger LOG = LoggerFactory.getLogger(SecretsManagerGroupNameAwareKeyValueFetcher.class);
+
+    private final SecretsManagerClient secretsClient;
+    private final ObjectMapper objectMapper;
+
+    /**
+     * @param secretsClient Secrets Client
+     * @param objectMapper Object Mapper
+     */
+    public SecretsManagerGroupNameAwareKeyValueFetcher(SecretsManagerClient secretsClient,
+                                                       ObjectMapper objectMapper) {
+        this.secretsClient = secretsClient;
+        this.objectMapper = objectMapper;
+    }
+
+    @Override
+    @NonNull
+    public Optional<Map<String, Map>> keyValuesByPrefix(@NonNull String prefix) {
+        Map<String, Map> result = new HashMap<>();
```

Comment:
Use a linked hash map incase order is important

Reply:
The class which calls this method manages the order of precedence for all the active env prefixes with the LinkedHashMap. So here it shouldn't be important.